diff --git a/src/prime_factors/primes.py b/src/prime_factors/primes.py
index e69de29..8c73348 100644
--- a/src/prime_factors/primes.py
+++ b/src/prime_factors/primes.py
@@ -1,2 +1,11 @@
 def generate(n: int):
-    return []
+    if n == 1:
+        return []
+    factors = []
+    divisor = 2
+    while n > 1:
+        while n % divisor == 0:
+            factors.append(divisor)
+            n //= divisor
+        divisor += 1
+    return factors
